### ISSUE TITLE
Consider upgrading from v120_xp to v141_xp 

### ISSUE THREAD

**Issue by ioncodes** [x64dbg/x64dbg#1506]
Consider to upgrade to Visual Studio 2017 to use latest toolsets.

**Comment by mrexodia** [comment]
There are no plans to upgrade as of now. Qt doesn't even provide packages for VS2017.

**Comment by mirh** [comment]
And it isn't at least [before](http://lists.qt-project.org/pipermail/releasing/2017-February/004471.html) [5.9](https://bugreports.qt.io/browse/QTBUG-57086#changehistory-903436). 
Though, at least Vs 2015 should be well supported.

**Comment by mrexodia** [comment]
Yeah I don't really see the point in switching. It only gives me extra
hassle because I have to convert the build server and old Qt plugins
probably won't work anymore...

**Comment by mrexodia** [comment]
@torusrxxx do you have a problem with upgrading to Visual Studio 2015?

**Comment by torusrxxx** [comment]
I don't see the point upgrading to Visual Studio 2015.

**Comment by mrexodia** [comment]
The issue is that Visual Studio 2013 is deprecated and you can no longer download it. Visual Studio 2019 will also release soon, but hopefully they allow downloading of the v141 toolset like in Visual Studio 2017. Basically the reason is that it becomes really difficult to contribute

**Comment by torusrxxx** [comment]
Ok. Are there any blockers of upgrading? Qt 5.11 doesn't support 2013 anymore.

**Comment by mrexodia** [comment]
I don't think there are any real blockers. I will also update Qt to the latest version and test to see if there are no new performance issues introduced. I just asked you in case you have to download vs 2015 :)

**Comment by torusrxxx** [comment]
Yes, you can upgrade x64dbg to use Visual Studio 2015.

**Comment by LFriede** [comment]
> The issue is that Visual Studio 2013 is deprecated and you can no longer download it. Visual Studio 2019 will also release soon, but hopefully they allow downloading of the v141 toolset like in Visual Studio 2017. Basically the reason is that it becomes really difficult to contribute

I was able to download VS 2013 with a free Microsoft Account. But it's difficult to setup Qt 5.6.x since it disappeared from the "Qt Maintenance Tools" and the installer.
It's possible to make it install old Qt versions by modifying http traffic, files are still there. I've written a small guide for that and leave the link here because I think it's helpful for contributors and plugin authors: https://gist.github.com/LFriede/3d4cde1bfdb211db4896a0a8b890660b

**Comment by mrexodia** [comment]
Actually all VS versions have support for 10 years. Anyway for development
people are using VS2019 and latest Qt without problems.

Plugins are working as well, I’m working on a better project template that
should work for all VS versions so that might be useful too.

Thanks for the guide though! I’ll see if I can add that to the wiki (which
actually has direct links to the Qt installers).

**Comment by LFriede** [comment]
> Thanks for the guide though! I’ll see if I can add that to the wiki (which
actually has direct links to the Qt installers).

I've read the wiki and used the linked installers previously, the problem is that they are bundled with Qt Creator and you can't install them in the same directory or add the old versions to an existing Qt Creator installation. That really annoyed me :smiley:

**Comment by torusrxxx** [comment]
x64dbg already transitioned to VS2022
